### PR TITLE
Add `@speechmatics/auth` package for JWT creation

### PR DESCRIPTION
**Pull request by mnemitz**
- Add `@speechmatics/auth` package which can be used to generate JWTs for batch, real-time and Flow
- Readme + links to documentation
- Updated examples

### CODE REVIEW

File: packages/auth/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export async function createSpeechmaticsJWT({
```
Comment (TudorCRL) [inline packages/auth/src/index.ts:1]: This will also need a region claim - can be eu, us, or au - and defaults to eu

**Review by TudorCRL — COMMENTED** [review]
mostly lgtm, just one small comment concerning regions

File: packages/auth/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}: {
+  type: 'batch' | 'rt' | 'flow';
+  apiKey: string;
+  clientRef?: string;
+  ttl?: number;
+  managementPlatformURL?: string;
+}): Promise<string> {
+  if (type === 'batch' && !clientRef) {
+    throw new Error(
+      'Must set the `client_ref` parameter when using temporary keys for batch transcription. See documentation at https://docs.speechmatics.com/introduction/authentication#batch-transcription',
```
Comment (anjz) [inline packages/auth/src/index.ts:16]: Do we want to enforce `client_ref` for batch jwt's?

Comment (mnemitz) [reply packages/auth/src/index.ts:20]: I brought this up in the SE channel to get their thoughts. It seems there might be one customer who is using JWTs without client_ref, but I'm unclear about the use-case.

I'm still inclined to enforce it here as a sane default. Worst case if they really need an escape hatch, they can always call the endpoint directly.

Comment (TudorCRL) [reply packages/auth/src/index.ts:20]: Sounds good to me

**Review by anjz — APPROVED** [review]

**Pull request merged**